python/ray/serve/_private/common.py

```diff
@@ -102,30 +105,53 @@ class DeploymentStatus(str, Enum):
     UPDATING = "UPDATING"
     HEALTHY = "HEALTHY"
     UNHEALTHY = "UNHEALTHY"
+    UPSCALING = "UPSCALING"
+    DOWNSCALING = "DOWNSCALING"
```

Comment:
> Actually, thinking about it a bit more, why do we even need this standalone state, provided that we already have the trigger specifying the intent for us making it clear that deployment is _updating_ b/c of auto-scaling

I think autoscaling should be it's own state(s). Updating means the configuration for a deployment has changed (modified by the user). For that reason, before this PR, the state of a deployment remains `HEALTHY` when a deployment is autoscaling. For more observability, we added explicit autoscaling states.

> Why do we need 2 states, can we just do it with 1 (AUTOSCALING)?

This is also for more observability, to show the direction the deployment is autoscaling in.

Reply:
State has to be reflective of the _state_ of the entity (application, deployment), while _trigger_ captures what actually provoked the state-transition.

We need to carefully balance these 2: state machine should be concise, representative and orthogonal (for ex, we'd not communicate whether we're up- or down-scaling using state, b/c it is not reflecting the state, but an action taken against it)

For ex, the question we need to answer how are UPDATING and AUTOSCALING states differ from the perspective of the entity itself (rather than external perspective when we factor in the trigger)?

Reply:
Synced offline, we'll be keeping the UPSCALING/DOWNSCALING states but making the status trigger more verbose.